addons/mail/static/src/components/composer/composer_tests.js

```diff
@@ -2098,6 +2098,54 @@ QUnit.test("mentioned partners should not be notified if they are not member of
     assert.verifySteps(['message_post'], "the message should be posted");
 });
 
+QUnit.test('does not crash when an attachment is removed before its upload starts', async function (assert) {
+    assert.expect(2);
```

Comment:
The test is good functionally, but the part with resolving the promise and canceling 2nd attachment during upload of 1st attachment is extremely technical. Name of test show how implementation-detail this test is. Would definitely put `[technical]` tag in that case. Clarify what are the technical assumptions that are being considered by the test, namely inputing 2 or more files for upload uploads the files sequentially (one at a time), and cancelling a file when it's hasn't start actual uploading should not crash.